csrc/compute_at_map.cpp

```diff
@@ -374,61 +374,63 @@ void IterDomainGraph::build(Fusion* fusion) {
       if (first_output_tv == nullptr) {
         first_output_tv = c_tv;
       } else {
-        // Map multi outputs of an expression to each other. c is current
-        // output, and f as first output. Keep consistent with the later section
-        // of producer and consumers. Which here producer is now "first output",
-        // and consumer is still consumer. One exception is how the
-        // domains left of CA positions are handled in the Parallel
-        // map. Those domains are not mapped in producer and consumer
-        // mappings as they do not share loops, but are mapped in the
-        // case of mapping multiple outputs since they do share the
-        // same loops.
-
-        NVF_ERROR(
-            c_tv->getMaybeRootDomain().size() ==
-                first_output_tv->getMaybeRootDomain().size(),
-            "Multiple outputs with mismatched dimensions is not supported. ",
-            "Only supported case is welford op where all outputs tvs have identical domains.");
-        // p->f, c->c
-        std::unordered_map<IterDomain*, IterDomain*> c2f_root_map;
-        for (const auto i :
-             c10::irange(first_output_tv->getMaybeRootDomain().size())) {
-          c2f_root_map.insert(std::make_pair(
-              c_tv->getMaybeRootDomain()[i],
-              first_output_tv->getMaybeRootDomain()[i]));
-        }
-
-        // Multi output mapping, outputs are required to have the same domain
-        // and same transformations, so they can be mapped in permissive/exact,
-        // and when within compute at position of getLoopDomain() in the
-        // parallel map.
-        auto replay_FasC = BestEffortReplay(
-            first_output_tv->getLoopDomain(),
-            c_tv->getLoopDomain(),
-            c2f_root_map);
-
-        // Map the entire replay map between the multiple
-        // consumers
-        auto c2f_disjoint_sets = replay_FasC.getIterDomainEquivalence();
-        for (const auto& disjoint_set : c2f_disjoint_sets.disjointSets()) {
-          if (disjoint_set->empty()) {
-            continue;
+        if (ir_utils::hasUniformSiblings(expr)) {
```

Comment:
nit: I think you can move this check to the beginning of this loop body, enclosing lines 374 and 375 as well.